src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1144,5 +1144,16 @@ public override int GetHashCode()
 
             return referenceCulture.CompareInfo.IsPrefix(this, value, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
         }
+
+        [Pure]
+        public Boolean StartsWith(Char value)
```

Comment:
I'll try to compare the assembly code generate. But you are right:

        /// This is a EE implemented function so that the JIT can recognise is specially
        /// and eliminate checks on character fetchs in a loop like:
        ///        for(int i = 0; i < str.Length; i++) str[i]
        /// The actually code generated for this will be one instruction and will be inlined.
        //
        // Spec#: Add postcondition in a contract assembly.  Potential perf problem.
        public extern int Length {
            [MethodImplAttribute(MethodImplOptions.InternalCall)]
            get;
        }